File: Week6/Day4/E2E_basics/steps/wiki_links_steps.py
from behave import given, when, then


@given(u'I am on the Wikipedia home page')
def step_impl(context):
    context.driver.get("https://www.wikipedia.org/")


@when(u'I click on the English link')
def step_impl(context):
    context.wiki_home.english().click()


@then(u'I should be on the English Wikipedia main page')
def step_impl(context):
    assert context.driver.title == "Wikipedia, the free encyclopedia"


@when(u'I click on the Spanish link')
def step_impl(context):
    context.wiki_home.spanish().click()


@then(u'I should be on the Spanish Wikipedia main page')
def step_impl(context):
    assert context.driver.title == "Wikipedia, la enciclopedia libre"


@when(u'i click on the Italian link')
def step_impl(context):
    context.wiki_home.italian().click()


@then(u'I should be on the Italian Wikipedia main page')
def step_impl(context):
    assert context.driver.title == "Wikipedia, l'enciclopedia libera"

File: Week9/Day3/src/test/java/com/revature/steps/WikiLinksSteps.java
package com.revature.steps;

import com.revature.runner.TestRunner;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.Assert;

public class WikiLinksSteps {

    @Given("I am on the Wikipedia home page")
    public void i_am_on_the_wikipedia_home_page() {
        // this step, despite being used in multiple feature files, only needs to be defined once
        TestRunner.driver.get("https://www.wikipedia.org/");
    }
    @When("I click the English link")
    public void i_click_the_english_link() {
        TestRunner.wikiHome.english.click();
    }
    @Then("I should be on the English home page")
    public void i_should_be_on_the_english_home_page() {
        String title = TestRunner.driver.getTitle();
        Assert.assertEquals("Wikipedia, the free encyclopedia", title);
    }

    @When("I click the Spanish link")
    public void i_click_the_spanish_link() {
        TestRunner.wikiHome.spanish.click();
    }
    @Then("I should be on the Spanish home page")
    public void i_should_be_on_the_spanish_home_page() {
        Assert.assertEquals("Wikipedia, la enciclopedia libre", TestRunner.driver.getTitle());

    }

    @When("I click the Italian link")
    public void i_click_the_italian_link() {
        TestRunner.wikiHome.italian.click();
    }
    @Then("I should be on the Italian home page")
    public void i_should_be_on_the_italian_home_page() {
        Assert.assertEquals("Wikipedia, l'enciclopedia libera", TestRunner.driver.getTitle());
    }

}
